kolibri/core/utils/csv.py

```diff
@@ -1,14 +1,67 @@
 import io
+import logging
 import re
+from contextlib import contextmanager
 from numbers import Number
 
+from django.core.files.storage import default_storage
 
-def open_csv_for_writing(filepath):
-    return io.open(filepath, "w", newline="", encoding="utf-8-sig")
+logger = logging.getLogger(__name__)
 
 
-def open_csv_for_reading(filepath):
-    return io.open(filepath, "r", newline="", encoding="utf-8-sig")
+def validate_open_csv_params(storage_filepath, local_filepath):
+    if storage_filepath is None and local_filepath is None:
+        raise ValueError("Either storage_filepath or local_filepath must be provided")
+    if storage_filepath and local_filepath:
+        raise ValueError(
+            "Only one of storage_filepath or local_filepath should be provided"
+        )
+
+
+@contextmanager
+def open_csv_for_writing(storage_filepath=None, local_filepath=None):
+
+    validate_open_csv_params(storage_filepath, local_filepath)
+
+    if storage_filepath:
+        # If the file does not exist, we need to create it and return it wrapped in a TextIOWrapper
+        with io.BytesIO() as f:
+            encoded_fh = io.TextIOWrapper(
+                f,
+                newline="",
+                encoding="utf-8-sig",
+                write_through=True,
+                line_buffering=True,
+            )
+            yield encoded_fh
+            encoded_fh.flush()
+            default_storage.save(storage_filepath, f)
```

Comment:
Looks like before we write here, we need to do an exists check and delete if so?

```
if default_storage.exists(storage_filepath):
    default_storage.delete(storage_filepath)
```
then do the save.